logic.py

```diff
@@ -81,9 +82,10 @@ def tell(self, sentence):
         self.clauses.extend(conjuncts(to_cnf(sentence)))
 
     def ask_generator(self, query):
-        "Yield the empty substitution if KB implies query; else nothing."
+        "Return the empty substitution {} if KB entails query; else return False."
         if tt_entails(Expr('&', *self.clauses), query):
-            yield {}
+            yield {}  # Why use yield when you are not returning a generator?
```

Comment:
For First Order Logic, yes. But for Propositional Logic, `ask_generator` will return either `{}` or `None`, which we can change to return simply `True` or `False` instead. I was wondering initially as to the original author's intention regarding this. But I think I get it now: this behavior of the function is tolerated to, maybe, maintain consistency.